.github/workflows/ci.yaml

```diff
@@ -57,6 +57,75 @@ jobs:
         run: nox -e ci_checks_max pytest_min
         timeout-minutes: 10
 
+  nox-cross-arch:
+    name: Cross-arch tests with nox
+    needs: ["nox"]
+    if: github.event_name == 'merge_group'
```

Comment:
Done

Reply:
I just saw your comment about this taking ~15m (I would say over 5m is too much). I think that's a bit too much to run on every PR. I guess the best next option would be to run it always, but inside the `run` check if it is a PR and if it is, just make it a NOP (ideally mark it as skipped but I'm not sure if a branch protection rule is considered fulfilled if the check was skipped).

This way we can require for the check to pass, and it will only run on the merge queue for PRs, and will be rejected from the queue if it fails. It will also run for pushes, and if we add it as a dependency for the `publish-docs` job, then a release will be blocked if the ARM tests fail.